### PR TITLE
Add perf pipeline

### PR DESCRIPTION
**Pull request by adiaaida**
This change will add a performance pipeline to run performance testing on internal builds. After this change is checked in, we will need to create the perf pipeline in AzDO.

This change enables performance testing per-commit on internal builds for Windows x64 and x86, and Ubuntu x64.

This change was tested as part of #25390, and has been tested in the performance repo to make sure the change works on internal runs.

This change will need to wait until https://github.com/dotnet/arcade/pull/3680 has been checked in and the arcade changes brought over to coreclr before we can enable perf testing (that change updates the default framework to netcoreapp5.0 and fixes a bug in the linux build).

Note: this change is only for ci testing. I have only enabled it for master, but we can discuss enabling it for release/3.0 until we ship (at which point, we will disable it). Enabling per-commit perf testing for release/3.0 will require us to set the framework to netcoreapp3.0.

### CODE REVIEW

File: eng/perf-job.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+### Test job
+
+### Each test job depends on a corresponding build job with the same
+### buildConfig and archType.
+
+jobs:
+- template: /eng/common/templates/job/performance.yml
+  parameters:
+    # Compute job name from template parameters
+    jobName: ${{ format('perfbuild_{0}_{1}_{2}', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
```
Comment (AaronRobinsonMSFT) [inline eng/perf-job.yml]: Please also create a `displayName` that doesn't have underscores - e.g. Perf Windows_NT x64 Release.

File: eng/perf-job.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  crossrootfsDir: ''
+  framework: netcoreapp5.0 # Specify the appropriate framework when running release branches (ie netcoreapp3.0 for release/3.0)
+
+### Test job
+
+### Each test job depends on a corresponding build job with the same
+### buildConfig and archType.
+
+jobs:
+- template: /eng/common/templates/job/performance.yml
```
Comment (AaronRobinsonMSFT) [inline eng/perf-job.yml]: We should only be running performance runs on Release builds. I don't know if we want to state that in the template or at instantiation time.

File: eng/perf-job.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    steps:
+    # Extra steps that will be passed to the performance template and run before sending the job to helix (all of which is done in the template)
+
+    # Download product build
+    - task: DownloadBuildArtifacts@0
+      displayName: Download product build
+      inputs:
+        buildType: current
+        downloadType: single
+        artifactName: ${{ format('{0}_{1}_{2}_build', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
```
Comment (AaronRobinsonMSFT) [inline eng/perf-job.yml]: I don't believe this is the artifact name any more.

File: eng/perf-job.yml (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply eng/perf-job.yml]: I see it is in the `perf.yml`. - Thanks.

File: eng/perf-job.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  archType: ''
+  osGroup: ''
+  osIdentifier: ''
+  readyToRun: true
+  helixQueues: ''
+  container: ''
+  crossrootfsDir: ''
+  framework: netcoreapp5.0 # Specify the appropriate framework when running release branches (ie netcoreapp3.0 for release/3.0)
+
+### Test job
```
Comment (jashook) [inline eng/perf-job.yml]: This seems like it does not fit as this isn't a test job?

Comment (adiaaida) [reply eng/perf-job.yml]: Copy paste from the test-job yaml

File: eng/perf-job.yml (hunk 3 above)
Comment (adiaaida) [reply eng/perf-job.yml]: I'll go look at what we do in test-job.yml and make sure it's the same.

File: eng/perf-job.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      # Public Windows Build Pool
+      ${{ if and(eq(parameters.osGroup, 'Windows_NT'), eq(variables['System.TeamProject'], 'public')) }}:
+        name: NetCorePublic-Pool
+        queue: BuildPool.Windows.10.Amd64.VS2017.Open
+
+    # Test job depends on the corresponding build job
+    dependsOn: ${{ format('build_{0}_{1}_{2}', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
+
+    # Run all steps in the container.
+    # Note that the containers are resources defined in azure-pipelines.yml
```
Comment (hoyosjs) [inline eng/perf-job.yml]: Stale comment. `azure-pipelines.yml` doesn't exist anymore.

File: eng/perf-job.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
+parameters:
+  buildConfig: ''
+  archType: ''
+  osGroup: ''
+  osIdentifier: ''
+  readyToRun: true
+  helixQueues: ''
+  container: ''
+  crossrootfsDir: ''
```
Comment (echesakovMSFT) [inline eng/perf-job.yml]: You can remove parameters that are not used by this job. For, example `crossrootfsDir`, `helixQueues`, `readyToRun`

**Comment by echesakovMSFT** [comment]
Do we have any plans to run this pipeline in public CI?

**Comment by adiaaida** [comment]
@echesakovMSFT No. Our perf queues are only available for internal builds. We do have an Open queue, but it is intended to be used for PR perf runs when we are finished implementing that support. The Open queue only has like 5 machines on it, so it is not setup to take per-commit workloads (the pr runs will be triggered by running an azp run command).

**Comment by adiaaida** [comment]
This change does not affect the arm64 job, which was abandoned. Going to merge.

**Pull request merged**